### ISSUE TITLE
The keepalived do not delete old virtual routes after reloading config if track_process is using

### ISSUE THREAD

**Issue by skydevil56** [acassen/keepalived#1218]
**Problem:**
The keepalived do not delete old virtual routes after reloading config if track_process is using.

**Informations:**

1. OS: Debian 9 x64 (Linux debian-9-x64 4.9.0-8-amd64 #1 SMP Debian 4.9.144-3.1 (2019-02-19) x86_64 GNU/Linux).
2. keepalived: 2.0.15 (git commit v2.0.14-53-g9de2817c+).

**Steps to reproduce:**
1. Create simple keepalived config /etc/keepalived/keepalived.conf:
```
root@sterragate:~# cat /etc/keepalived/keepalived.conf
vrrp_track_process getcrls_daemon {
  process "getcrls_daemon"
  quorum 1
  delay 5
}
vrrp_instance eth0_2 {
  track_process {
    getcrls_daemon
  }
  interface eth0
  virtual_routes {
    src 100.100.100.100 2.2.2.2/32 via 100.100.100.1 dev eth0
  }
  virtual_ipaddress {
    100.100.100.100/24
  }
  priority 100
  advert_int 1
  virtual_router_id 2
}
```
2. Start the keepalvied and check that virtual route exist:
```
root@sterragate:~# systemctl restart keepalived.service
root@sterragate:~# ps -Af | grep keepalived
root       2191      1  0 15:45 ?        00:00:00 /usr/sbin/keepalived -d -P -S 7 -x -g
root       2192   2191  0 15:45 ?        00:00:00 /usr/sbin/keepalived -d -P -S 7 -x -g
root       2199   1939  0 15:45 pts/0    00:00:00 grep keepalived
root@sterragate:~# ip r
default via 10.0.0.1 dev eth0
2.2.2.2 via 100.100.100.1 dev eth0 proto 112 src 100.100.100.100            < --- virtual route
10.0.0.0/16 dev eth0 proto kernel scope link src 10.0.238.206
100.100.100.0/24 dev eth0 proto kernel scope link src 100.100.100.100
192.168.100.0/24 dev eth1 proto kernel scope link src 192.168.100.1

```
3. Change virtual route ( from 'src 100.100.100.100 2.2.2.2/32 via 100.100.100.**1** dev eth0' to 'src 100.100.100.100 2.2.2.2/32 via 100.100.100.**2** dev eth0') in config and reload the keepalvied:

```
root@sterragate:~# cat /etc/keepalived/keepalived.conf
vrrp_track_process getcrls_daemon {
  process "getcrls_daemon"
  quorum 1
  delay 5
}
vrrp_instance eth0_2 {
  track_process {
    getcrls_daemon
  }
  interface eth0
  virtual_routes {
    src 100.100.100.100 2.2.2.2/32 via 100.100.100.2 dev eth0
  }
  virtual_ipaddress {
    100.100.100.100/24
  }
  priority 100
  advert_int 1
  virtual_router_id 2
}
root@sterragate:~# kill -s $(keepalived --signum=RELOAD) $(cat /var/run/keepalived.pid)
root@sterragate:~# ip r
default via 10.0.0.1 dev eth0
2.2.2.2 via 100.100.100.2 dev eth0 proto 112 src 100.100.100.100       < ---  new virtual route
2.2.2.2 via 100.100.100.1 dev eth0 proto 112 src 100.100.100.100       < ---  old virtual route
10.0.0.0/16 dev eth0 proto kernel scope link src 10.0.238.206
100.100.100.0/24 dev eth0 proto kernel scope link src 100.100.100.100
192.168.100.0/24 dev eth1 proto kernel scope link src 192.168.100.1
```

**Comment by pqarmitage** [comment]
The problem was due to keepalived segfaulting while cleaning up before the reload, and consequently when it was restarted by the parent keepalived process, it had no knowledge of the old route (since it was no longer in the config) and so didn't delete it.

This issue is resolved by commit aaa17ee.

**Comment by skydevil56** [comment]
Thank you very much.
